mitodl/ocw-studio
# Added WebsiteContent filename and dirpath fields

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #247 
Closes #251 

#### What's this PR do?
- Adds `WebsiteContent.filename` field
- Adds `WebsiteContent.dirpath` field
- Adds data migration to populate both of the above fields
- Adds logic to auto-generate a filename value from the title, and attempt to generate a new filename if there's a conflict
- Updates OCW course site import to populate the new fields

#### How should this be manually tested?
**Testing filename**: 
- Use the app to create new content for the same site and type (i.e.: pick a single website and add several objects for "Page"), and using the same value for the title. The `WebsiteContent` objects that are created should have a non-null `filename` value, and they should be equal to the slug-ified title with a numerical suffix if necessary
- Check the `WebsiteContent` records for imported OCW course sites. The filename value should be the `content_filepath` value without the dirpath and file extension

**Testing dirpath**: 
- Check the `WebsiteContent` records for imported OCW course sites. The dirpath value should be the `content_filepath` value without the filename and file extension

#### Any background context you want to provide?
- Finalizing the logic for determining the destination filepath for a given `WebsiteContent` object and getting rid of `content_filepath` will be done in a separate issue.
